torch/jit/__init__.py

```diff
@@ -369,14 +369,17 @@ def _script_graph(fn, _frames_up=0):
 
 def script(fn, _frames_up=0):
     graph = _script_graph(fn, _frames_up=_frames_up + 1)
-    return torch._C.GraphExecutor(graph, True)
+    fn_attributes = {i: getattr(fn, i) for i in functools.WRAPPER_ASSIGNMENTS}
+    Wrapper = type('GraphExecutorWrapper', (torch._C.GraphExecutor, ), fn_attributes)
+    return Wrapper(graph, True)
 
 
-ScriptMethodStub = namedtuple('ScriptMethodStub', ('resolution_callback', 'ast'))
+ScriptMethodStub = namedtuple('ScriptMethodStub', ('resolution_callback', 'ast', 'wrap'))
 
 
 def script_method(fn):
-    return ScriptMethodStub(createResolutionCallback(frames_up=1), get_jit_ast(fn))
+    wrap = functools.wraps(fn)(lambda: None)
+    return ScriptMethodStub(createResolutionCallback(frames_up=1), get_jit_ast(fn), wrap)
```

Comment:
It would be much nicer to just store `functools.wraps(fn)`, and later just do `wrap(...)` instead of `functools.wraps(wrap)(...)` again.

Alternatively it might be a good idea to just stash the original method, and call `functools.wraps(original_method)(<value form __getattr__>)`

Reply:
I stash the original method. This contains more information of the original method and might be useful for later changes if we need to add some new feature to it.